pxr/imaging/hdx/shaders/renderPass.glslfx

```diff
@@ -176,11 +239,95 @@ void RenderOutput(vec4 Peye, vec3 Neye, vec4 color, vec4 patchCoord)
     int instanceId = GetDrawingCoord().instanceIndex[0];
     instanceIdOut = IntToVec4(instanceId);
 
-    elementIdOut = IntToVec4(GetElementID());
-    edgeIdOut = IntToVec4(GetPrimitiveEdgeId());
-    pointIdOut = IntToVec4(GetPointId());
+    int elementId = GetElementID();
+    int edgeId = GetPrimitiveEdgeId();
+    int pointId = GetPointId();
+
+    elementIdOut = IntToVec4(elementId);
+    edgeIdOut = IntToVec4(edgeId);
+    pointIdOut = IntToVec4(pointId);
     
     neyeOut = IntToVec4(hd_vec4_2_10_10_10_set(vec4(Neye,0)));
+
+    // add the item to the pick buffer, but only unique
```

Comment:
And to get a bit deeper into this; I'm not up-to-date on how expensive an ATOMIC_LOAD will be if nobody's writing to the SSBO (because we're not deep picking), but if this might be a performance concern we could replace it with one of the following:

1. Removing the "pickBuffer" binding when we're not deep picking, which should be pretty fast.
2. Adding a uniform to tell us whether to run the deep picking code.

Would appreciate any information about whether you expect/have seen this to be a performance concern or a non-issue.

Reply:
Got around to running my own timings and the ATOMIC_LOAD is a non-issue, at least on the hardware I have access to, so I think it's fine until someone comes up with a hardware configuration where it's problematic.